docs/dxp/latest/en/developing-applications/core-frameworks/configurable-application/ddm-form-annotations/resources/liferay-v1d9.zip/v1d9-impl/src/main/java/com/acme/v1d9/internal/configuration/admin/definition/V1D9ConfigurationForm.java

```diff
@@ -0,0 +1,60 @@
+package com.acme.v1d9.internal.configuration.admin.definition;
+
+import com.liferay.dynamic.data.mapping.annotations.DDMForm;
+import com.liferay.dynamic.data.mapping.annotations.DDMFormField;
+import com.liferay.dynamic.data.mapping.annotations.DDMFormLayout;
+import com.liferay.dynamic.data.mapping.annotations.DDMFormLayoutColumn;
+import com.liferay.dynamic.data.mapping.annotations.DDMFormLayoutPage;
+import com.liferay.dynamic.data.mapping.annotations.DDMFormLayoutRow;
+
+@DDMForm
+@DDMFormLayout(
+	paginationMode = com.liferay.dynamic.data.mapping.model.DDMFormLayout.SINGLE_PAGE_MODE,
+	value = {
+		@DDMFormLayoutPage(
+			{
+				@DDMFormLayoutRow(
+					{
+						@DDMFormLayoutColumn(size = 6, value = "text"),
+						@DDMFormLayoutColumn(size = 6, value = "checkbox")
+					}
+				),
+				@DDMFormLayoutRow(
+					{
+						@DDMFormLayoutColumn(size = 6, value = "select"),
+						@DDMFormLayoutColumn(size = 6, value = "numeric")
+					}
+				),
+				@DDMFormLayoutRow(
+					{@DDMFormLayoutColumn(size = 12, value = "date")}
+				)
+			}
+		)
+	}
+)
+public interface V1D9ConfigurationForm {
+
+	@DDMFormField(label = "Checkbox", properties = "showAsSwitcher=true")
+	public boolean checkbox();
+
+	@DDMFormField(label = "Date", type = "date")
+	public String date();
+
+	@DDMFormField(
+		label = "Numeric", properties = "placeholder=%Seconds",
```

Comment:
@jhinkey Something weird I found is that if this '%' is removed, the whole thing crashes :man-shrugging:

Reply:
OK. In liferay-portal, the `%` prefix seems to be the convention. :-)

`git grep "placeholder=" $(git grep -l "@DDMFormField" *.java)`